rpm/src/main/java/org/eclipse/packager/rpm/RpmTagValue.java

```diff
@@ -118,19 +117,44 @@ public Optional<Long> asLong() {
             return Optional.empty();
         }
 
-        if (this.value instanceof Long) {
-            return Optional.of(((Long) this.value).longValue());
+        if (value instanceof Number) {
```

Comment:
I am not sure about this change. First of all, it's not aligned with the other (int for example). But also because it won't work with the array variant.

One way to deal with this could be to add a `asNumber` variant. Which would still need some strategy for the array case I guess.

Reply:
@ctron This code actually came from preexisting code in `RpmInformations::asLong` that I merged into `RpmTagValue::asLong`. I don't think I actually changed anything. However, I reverted all changes to this file now except for the absolute minimum, so please look again.

In general, I do not like automatically converting between `Integer` and `Long` and I'd like to get rid of it in a different pull request. Java 8 has `Integer::toUnsignedLong` which we can use for `uint32` where needed, but this doesn't actually require changing the data type. We should just change it at the `RpmInformations` level, but I think we should make that an even higher level, i.e., store an `Instant` instead of a `Long` there.